### PR TITLE
deprecate TO_FILE and make all private methods private

### PR DESCRIPTION
**Pull request by grosser**
@noahd1 @jonathancadepowers

### CODE REVIEW

File: spec/lib/formatter_spec.rb [hunk 1]
```diff
@@ -107,9 +107,13 @@ module CodeClimate::TestReporter
...
     describe '#short_filename' do
+      def call(*args)
+        formatter.send(:short_filename, *args)
+      end
+
       it 'should return the filename of the file relative to the SimpleCov root' do
-        expect(formatter.short_filename('file1')).to eq('file1')
-        expect(formatter.short_filename("#{::SimpleCov.root}/file1")).to eq('file1')
+        expect(call('file1')).to eq('file1')
+        expect(call("#{::SimpleCov.root}/file1")).to eq('file1')
```
Comment (noahd1) [inline spec/lib/formatter_spec.rb]: I think I prefer the explicit style. This is harder for me to read.

File: lib/code_climate/test_reporter/formatter.rb [hunk 2]
```diff
@@ -98,7 +99,13 @@ def round(numeric, precision)
         Float(numeric).round(precision)
       end
 
-      private
+      def write_to_file?
+        return true if tddium? || ENV["CODECLIMATE_TO_FILE"]
+        if ENV["TO_FILE"]
+          warn "TO_FILE is deprecated, use CODECLIMATE_TO_FILE"
```
Comment (noahd1) [inline lib/code_climate/test_reporter/formatter.rb]: Where is `warn` defined?

File: lib/code_climate/test_reporter/formatter.rb [hunk 3]
```diff
@@ -98,7 +99,13 @@ def round(numeric, precision)
...
+          true
```
Comment (noahd1) [inline lib/code_climate/test_reporter/formatter.rb]: It might be simpler / easier to read if you separate the deprecation warning from the return value of this method. 

That is this method could return, more simply without early returns the result of `tddium? || ENV["CODECLIMATE_TO_FILE"] || ENV["TO_FILE"]`. Before that you could print a deprecation warning.

File: spec/lib/formatter_spec.rb (hunk 1 above)
Comment (noahd1) [inline spec/lib/formatter_spec.rb]: Are these spec changes related to the changes in this commit? They don't seem to be.

File: lib/code_climate/test_reporter/formatter.rb [hunk 4]
```diff
@@ -36,6 +36,8 @@ def format(result)
         false
       end
 
+      private
```
Comment (noahd1) [inline lib/code_climate/test_reporter/formatter.rb]: Style note -- this repo uses a rather unconventional, but consistent "outdented" `private`. So you'll want to push this back a tab.

File: lib/code_climate/test_reporter/formatter.rb (hunk 2 above)
Comment (grosser) [inline lib/code_climate/test_reporter/formatter.rb]: warn is in kernel and basically is a `$stderr.puts`

File: spec/lib/formatter_spec.rb (hunk 1 above)
Comment (grosser) [inline spec/lib/formatter_spec.rb]: un-dry-ed it

**Comment by grosser** [comment]
good to go ?

File: lib/code_climate/test_reporter/formatter.rb (hunk 3 above)
Comment (grosser) [inline lib/code_climate/test_reporter/formatter.rb]: changed

File: lib/code_climate/test_reporter/formatter.rb [hunk 5]
```diff
@@ -36,6 +36,8 @@ def format(result)
         false
       end
 
+    private
```
Comment (grosser) [inline lib/code_climate/test_reporter/formatter.rb]: fyi was not outdetend before

File: lib/code_climate/test_reporter/formatter.rb (hunk 4 above)
Comment (grosser) [inline lib/code_climate/test_reporter/formatter.rb]: changed

File: spec/lib/formatter_spec.rb [hunk 6]
```diff
@@ -126,8 +126,8 @@ module CodeClimate::TestReporter
         end
 
         it 'should include the path prefix if set' do
-          expect(formatter.short_filename('file1')).to eq('custom/file1')
-          expect(formatter.short_filename("#{::SimpleCov.root}/file1")).to eq('custom/file1')
+          expect(formatter.send(:short_filename, 'file1')).to eq('custom/file1')
+          expect(formatter.send(:short_filename, "#{::SimpleCov.root}/file1")).to eq('custom/file1')
```
Comment (noahd1) [inline spec/lib/formatter_spec.rb]: I didn't understand before why you needed to change this spec  -- now I see it's because of the `private` keyword. 

Generally however, it's a bit of an anti-pattern to test private methods. Do you just want to leave the method public for now and leave the spec alone? It's either that, or refactor in order to test this as a public method on a different object.

**Comment by noahd1** [comment]
Looking better, thank you -- one last comment.

File: spec/lib/formatter_spec.rb (hunk 6 above)
Comment (grosser) [inline spec/lib/formatter_spec.rb]: It's also an anti-pattern to have methods public that are not public :)
Atm it's 'incorrect' code and 'incorrect' tests and changing it to
'correct' code and 'incorrect' tests is somehow bad ?

changed it back to be 'incorrect' code, since this was supposed to be a
simple fix and not re-architecting :)

Comment (noahd1) [inline spec/lib/formatter_spec.rb]: I happen to agree with you and didn't intend for you to do any re-architecture. Just wanted to add the least amount of code to get the job done. Thanks for your contribution.
